### PR TITLE
Add lint for GitHub Actions workflow files

### PR DESCRIPTION
**Pull request by cgdolan**
PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Review by spencerdrak — APPROVED** [review]

**Comment by spencerdrak** [comment]
This is slick! Nice @cgdolan !

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -100,3 +100,11 @@ jobs:
           opam exec -- pre-commit run --verbose --all lint-ocaml
           # to debug errors in pre-commit, use instead:
           # opam exec -- pre-commit run --verbose --all lint-ocaml || cat /root/.cache/pre-commit/pre-commit.log
+
+  github-actions:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: reviewdog/action-actionlint@v1
```
Comment (aryx) [inline .github/workflows/lint.yml:108]: Nice! Now we need to port those actionlint checks in Semgrep!

**Review by aryx — APPROVED** [review]

File: .github/workflows/lint.yml (hunk 1 above)
Comment (aryx) [reply .github/workflows/lint.yml:108]: Maybe you can remove the TODO? comment now next to the name: lint at the top.

**Review by aryx — APPROVED** [review]

File: .github/workflows/lint.yml (hunk 1 above)
Comment (cgdolan) [reply .github/workflows/lint.yml:108]: Thanks for the suggestion, I'll consider this for a follow up PR.

**Pull request merged**